File: Leetcode 2009. Minimum Number of Operations to Make Array Continuous/lc 2009.java
class Solution {
    public int minOperations(int[] nums) {
        HashSet<Integer> hs=new HashSet<>();
        for(int i:nums){
            hs.add(i);
        }
        Integer[] arr=new Integer[hs.size()];
        hs.toArray(arr);
        Arrays.sort(arr);
        // for(int i:arr){
        //     System.out.println(i);
        // }
        
        int min=Integer.MAX_VALUE;
        int r=0;
        for(int l=0;l<arr.length;++l){
            while(r<arr.length && arr[r]<arr[l]+nums.length){
                r+=1;
            }
            int window=r-l;
            min=Math.min(min,nums.length-window);
            
            if(r==arr.length) break;
        }
        return min;
        

    }
}

File: Leetcode 2009. Minimum Number of Operations to Make Array Continuous/lc 2009.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        int len=nums.size();

        set<int> st;
        for(int i=0;i<nums.size();i++){
            st.emplace(nums[i]);
        }
        vector<int> arr(st.size());
        copy(st.begin(),st.end(),arr.begin());

        int minn=INT_MAX;
        int r=0;

        for(int l=0;l<arr.size();l++){
            while(r<arr.size() && arr[r]<arr[l]+len){
                r++;
            }
            int window=r-l;
            minn=min(minn,len-window);
            if(r==arr.size()) break;
        }
        return minn;

    }
};